app/src/main/java/com/github/sdpcoachme/auth/LoginActivity.kt

```diff
@@ -127,42 +149,109 @@ class LoginActivity : ComponentActivity() {
                     ) {
                         // Need to pass padding to child node
                             innerPadding ->
-                        AuthenticationForm(modifier = Modifier.padding(innerPadding))
+                        AuthenticationForm(modifier = Modifier.padding(innerPadding), launchGoogleSignIn = launchGoogleSignIn)
                     }
                 }
             }
         }
     }
 
     @Composable
-    fun AuthenticationForm(modifier: Modifier = Modifier) {
+    fun AuthenticationForm(modifier: Modifier = Modifier, launchGoogleSignIn: MutableState<Boolean>) {
 
         var showUI by remember { mutableStateOf(false) }
 
         LaunchedEffect(true) {
             showUI = displayUI.await()
         }
 
+        // Coroutine responsible for launching (relaunching at cancel) the Google sign in
+        LaunchedEffect(key1 = launchGoogleSignIn.value) {
+            if (launchGoogleSignIn.value) {
+                // Wait 5 seconds before launching the sign in
+                // to display the login screen to the user
+                withContext(Dispatchers.IO) {
+                    sleep(5000)
```

Comment:
This has been done intentionally so that users coming to this interface after logging out have a clear cut.
Also, if it is shorter, it gives the user less time to read the text.